product_packaging_container_deposit/models/container_deposit_order_mixin.py

```diff
@@ -41,36 +46,63 @@ def update_order_container_deposit_quantity(self):
             deposit_container_qties = (
                 lines_to_comp_deposit._get_order_lines_container_deposit_quantities()
             )
+            values_lst = []
             for line in self[self._get_order_line_field()]:
                 if not line.is_container_deposit:
                     continue
                 qty, qty_dlvd_rcvd = deposit_container_qties.pop(
                     line["product_id"], [False, False]
                 )
                 if not qty:
+                    new_vals = {
+                        line._get_product_qty_field(): 0,
+                    }
                     if order.state == "draft":
-                        line.unlink()
-                    else:
-                        line.write(
-                            {
-                                line._get_product_qty_field(): 0,
-                            }
+                        # values_lst.append(Command.delete(line.id))
+                        line_ids_to_delete.append(line.id)
+                        # TODO: check if it is needed for UI only
+                        new_vals["name"] = _("[DEL] %(name)s", name=line.name)
+                    # else:
+                    values_lst.append(
+                        Command.update(
+                            line.id,
+                            new_vals,
                         )
+                    )
+
                 else:
-                    line.write(
-                        {
-                            line._get_product_qty_field(): qty,
-                            line._get_product_qty_delivered_received_field(): qty_dlvd_rcvd,
-                        }
+                    values_lst.append(
+                        Command.update(
+                            line.id,
+                            {
+                                line._get_product_qty_field(): qty,
+                                line._get_product_qty_delivered_received_field(): qty_dlvd_rcvd,
+                            },
+                        )
                     )
-            values_lst = []
             for product in deposit_container_qties:
                 if deposit_container_qties[product][0] > 0:
                     values = order.prepare_deposit_container_line(
                         product, deposit_container_qties[product][0]
                     )
                     values_lst.append(Command.create(values))
             order.write({self._get_order_line_field(): values_lst})
+        # Schedule line to delete after commit to avoid caching issue w/ UI
```

Comment:
Don't you get the issue on next call from UI ?
Seems safer to me to keep the lines but set quantity to 0

Reply:
Saw a demo, does the job !